Dockerfile

```diff
@@ -1,15 +1,23 @@
+ARG ALPINE_VERSION=3.12
+ARG NODE_VERSION=15
+ARG NGINX_VERSION=1.19.6
+
 # multistage - builder image
-FROM node:alpine AS builder
+FROM node:${NODE_VERSION}-alpine${ALPINE_VERSION} AS builder
 WORKDIR /app/Lightspeed-react
-COPY . .
+# Layer caching for node modules
+COPY package.json package-lock.json ./
 RUN npm install
-
-# configure ip, hardcoded to webrtc container address (8080) for now
-RUN sed -i "s|stream.gud.software|localhost|g" public/config.json
+COPY . .
 
 # build it
 RUN npm run build
 
 # runtime image
-FROM nginx:stable
-COPY --from=builder /app/Lightspeed-react/build /usr/share/nginx/html
+FROM nginx:${NGINX_VERSION}-alpine
```

Comment:
should be `FROM nginx:${NGINX_VERSION}-alpine AS lightspeed-react` , appending the "target". In the future there may be a subsequent build stage (eg. for collecting test results) that don't comprise the actual container image.

Reply:
The final image shouldn't contain a target, there is no point for it. And other stages should be placed before with a target name, I have never seen a use case to have non final stage after the final image description. Let me know if I'm missing something.

Reply:
It's possible but probably over-optimization - so not needed at this stage.